Please don't impose Werror on downstream users

`Werror` is great for development and CI, but is not appropriate for users of your library. New compiler versions frequently bring out new warnings or refine their heuristics. When this breaks the build, all downstream can reasonably do is blindly disable Werror anyway.

Fresh project using `plutus-tx` (and thus, indirectly, `cardano-crypto-class`) fails with GHC 9.4.6 with:

```
[ 6 of 48] Compiling Cardano.Crypto.Hash.Class ( src/Cardano/Crypto/Hash/Class.hs, dist/build/Cardano/Crypto/Hash/Clas>

src/Cardano/Crypto/Hash/Class.hs:173:38: error: [-Wdeprecations, -Werror=deprecations]
    In the use of ‘serializeEncoding'’
    (imported from Cardano.Binary, but defined in cardano-binary-1.6.0.0-9zN4iiyMBUGJoTfCIR8U3A:Cardano.Binary.Seriali>
    Deprecated: "Use `serialize'` instead, since `Encoding` has `ToCBOR` instance"
    |
173 | hashWithSerialiser toEnc = hashWith (serializeEncoding' . toEnc)
    |                                      ^^^^^^^^^^^^^^^^^^
```

--- a/base-deriving-via/CHANGELOG.md
+++ b/base-deriving-via/CHANGELOG.md
@@ -1,4 +1,8 @@
 ## 0.1.0.1
 
+* Remove `development` flag: #372
+
+## 0.1.0.1
+
 * Initial release
 

--- a/base-deriving-via/base-deriving-via.cabal
+++ b/base-deriving-via/base-deriving-via.cabal
@@ -1,7 +1,7 @@
 cabal-version:       >=1.10
 
 name:                base-deriving-via
-version:             0.1.0.1
+version:             0.1.0.2
 synopsis:            A general hook newtype for use with deriving via
 license:             Apache-2.0
 license-files:
@@ -12,11 +12,6 @@ maintainer:          operations@iohk.io
 copyright:           IOHK
 build-type:          Simple
 extra-source-files:  CHANGELOG.md
-
-flag development
-    description: Disable `-Werror`
-    default: False
-    manual: True
 
 library
   default-language:     Haskell2010
@@ -31,9 +26,6 @@ library
     -Widentities
     -Wredundant-constraints
     -Wmissing-export-lists
-  if (!flag(development))
-    ghc-options:
-      -Werror
 
   exposed-modules:
                         Data.DerivingVia

--- a/cabal.project
+++ b/cabal.project
@@ -34,3 +34,5 @@ packages:
 -- Ensures colourized output from test runners
 test-show-details: direct
 
+program-options
+  ghc-options: -Werror

--- a/cardano-binary/CHANGELOG.md
+++ b/cardano-binary/CHANGELOG.md
@@ -2,6 +2,7 @@
 
 ## 1.7.0.0
 
+* Remove `development` flag: #372
 * Remove `To/FromCBOR` instances for `NominalDiffTime`, since they did rounding.  Newly
   added functions `encodeNominalDiffTimeMicro`/`decodedNominalDiffTimeMicro` can be used
   to recover previous behavior. Correct instances that do not perform any rounding will

--- a/cardano-binary/cardano-binary.cabal
+++ b/cardano-binary/cardano-binary.cabal
@@ -16,11 +16,6 @@ build-type:          Simple
 extra-source-files:  README.md
                      CHANGELOG.md
 
-flag development
-    description: Disable `-Werror`
-    default: False
-    manual: True
-
 common base                         { build-depends: base                             >= 4.14       && < 4.17     }
 
 common project-config
@@ -33,9 +28,6 @@ common project-config
                         -Wpartial-fields
                         -Wredundant-constraints
                         -Wunused-packages
-
-  if (!flag(development))
-    ghc-options:        -Werror
 
 library
   import:               base, project-config

--- a/cardano-binary/test/CHANGELOG.md
+++ b/cardano-binary/test/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 1.4.0.1
+
+* Remove `development` flag: #372
+
 ## 1.4.0.0
 
 * Remove `Test.Cardano.Binary.Drop` module

--- a/cardano-binary/test/cardano-binary-test.cabal
+++ b/cardano-binary/test/cardano-binary-test.cabal
@@ -1,7 +1,7 @@
 cabal-version: 2.2
 
 name:                cardano-binary-test
-version:             1.4.0.0
+version:             1.4.0.1
 synopsis:            Test helpers from cardano-binary exposed to other packages
 description:         Test helpers from cardano-binary exposed to other packages
 license:             MIT
@@ -12,11 +12,6 @@ copyright:           2019-2021 IOHK
 category:            Currency
 build-type:          Simple
 extra-source-files:  CHANGELOG.md
-
-flag development
-    description: Disable `-Werror`
-    default: False
-    manual: True
 
 common base                         { build-depends: base >= 4.14 && < 4.17 }
 
@@ -30,9 +25,6 @@ common project-config
                         -Wpartial-fields
                         -Wredundant-constraints
                         -Wunused-packages
-
-  if (!flag(development))
-    ghc-options:         -Werror
 
 library
   import:               base, project-config

--- a/cardano-crypto-class/CHANGELOG.md
+++ b/cardano-crypto-class/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 2.1.0.1
+
+* Remove `development` flag: #372
+
 ## 2.1.0.0
 
 * Fixed the name `encodedSignKeyDESIGNSizeExpr` -> `encodedSignKeyDSIGNSizeExpr`

--- a/cardano-crypto-class/cardano-crypto-class.cabal
+++ b/cardano-crypto-class/cardano-crypto-class.cabal
@@ -1,6 +1,6 @@
 cabal-version:      2.2
 name:               cardano-crypto-class
-version:            2.1.0.0
+version:            2.1.0.1
 synopsis:
   Type classes abstracting over cryptography primitives for Cardano
 
@@ -20,11 +20,6 @@ build-type:         Simple
 extra-source-files: README.md
                     CHANGELOG.md
 
-flag development
-  description: Disable `-Werror`
-  default:     False
-  manual:      True
-
 flag secp256k1-support
     description: Enable support for functions from libsecp256k1. Requires
                  a recent libsecp256k1 with support for Schnorr signatures.
@@ -39,9 +34,6 @@ common project-config
     -Wall -Wcompat -Wincomplete-record-updates
     -Wincomplete-uni-patterns -Wpartial-fields -Wredundant-constraints
     -Wunused-packages
-
-  if !flag(development)
-    ghc-options: -Werror
 
 library
   import:            base, project-config

--- a/cardano-crypto-praos/cardano-crypto-praos.cabal
+++ b/cardano-crypto-praos/cardano-crypto-praos.cabal
@@ -1,7 +1,7 @@
 cabal-version: 2.2
 
 name:                cardano-crypto-praos
-version:             2.1.1.0
+version:             2.1.1.1
 synopsis:            Crypto primitives from libsodium
 description:         VRF (and KES, tba) primitives from libsodium.
 license:             Apache-2.0
@@ -37,11 +37,6 @@ extra-source-files:    cbits/crypto_vrf.h
                        cbits/private/fe_51/base2.h
                        cbits/private/fe_51/fe.h
 
-flag development
-    description: Disable `-Werror`
-    default: False
-    manual: True
-
 flag external-libsodium-vrf
     description: Rely on a special libsodium fork containing the VRF code.
                  Otherwise expect a normal unaltered system libsodium, and
@@ -61,9 +56,6 @@ common project-config
                         -Wpartial-fields
                         -Wredundant-constraints
                         -Wunused-packages
-
-  if (!flag(development))
-    ghc-options:        -Werror
 
 library
   import:               base, project-config

--- a/cardano-crypto-tests/CHANGELOG.md
+++ b/cardano-crypto-tests/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 2.1.0.1
+
+* Remove `development` flag: #372
+
 ## 2.1.0.0
 
 * Addition of `DSIGN` benchmarks. New modules:

--- a/cardano-crypto-tests/cardano-crypto-tests.cabal
+++ b/cardano-crypto-tests/cardano-crypto-tests.cabal
@@ -1,7 +1,7 @@
 cabal-version: 2.2
 
 name:                cardano-crypto-tests
-version:             2.1.0.0
+version:             2.1.0.1
 synopsis:            Tests for cardano-crypto-class and -praos
 description:         Tests for cardano-crypto-class and -praos
 license:             Apache-2.0
@@ -15,11 +15,6 @@ category:            Currency
 build-type:          Simple
 extra-source-files:  README.md
                      CHANGELOG.md
-
-flag development
-    description: Disable `-Werror`
-    default: False
-    manual: True
 
 flag secp256k1-support
     description: Enable support for functions from libsecp256k1. Requires
@@ -39,9 +34,6 @@ common project-config
                         -Wpartial-fields
                         -Wredundant-constraints
                         -Wunused-packages
-
-  if (!flag(development))
-    ghc-options:        -Werror
 
 library
   import:               base, project-config

--- a/cardano-slotting/CHANGELOG.md
+++ b/cardano-slotting/CHANGELOG.md
@@ -2,6 +2,7 @@
 
 ## 0.1.1.0
 
+* Addition of `ToJSON`/`FromJSON` instances for `BlockNo`
 * Addition of `ToJSON`/`FromJSON` instances for `SystemStart`
 * Addition of `ToCBOR`/`FromCBOR` instances for `RelativeTime` and `SlotLength`
 

--- a/cardano-slotting/CHANGELOG.md
+++ b/cardano-slotting/CHANGELOG.md
@@ -2,9 +2,11 @@
 
 ## 0.1.1.0
 
-* Addition of `ToJSON`/`FromJSON` instances for `BlockNo`
-* Addition of `ToJSON`/`FromJSON` instances for `SystemStart`
-* Addition of `ToCBOR`/`FromCBOR` instances for `RelativeTime` and `SlotLength`
+* Addition of `ToJSON` instances for `WithOrigin`.
+* Addition of `ToJSON`/`FromJSON` instances for:
+  * `BlockNo`
+  * `SystemStart`
+  * `RelativeTime` and `SlotLength`
 
 ## 0.1.0.1
 

--- a/cardano-slotting/CHANGELOG.md
+++ b/cardano-slotting/CHANGELOG.md
@@ -2,6 +2,7 @@
 
 ## 0.1.1.0
 
+* Remove `development` flag: #372
 * Addition of `ToJSON` instances for `WithOrigin`.
 * Addition of `ToJSON`/`FromJSON` instances for:
   * `BlockNo`

--- a/cardano-slotting/cardano-slotting.cabal
+++ b/cardano-slotting/cardano-slotting.cabal
@@ -13,11 +13,6 @@ copyright:           IOHK
 build-type:          Simple
 extra-source-files:  CHANGELOG.md
 
-flag development
-    description: Disable `-Werror`
-    default: False
-    manual: True
-
 common base                         { build-depends: base                             >= 4.14       && < 4.17     }
 
 common project-config
@@ -28,9 +23,6 @@ common project-config
                         -Wincomplete-uni-patterns
                         -Wredundant-constraints
                         -Wunused-packages
-
-  if (!flag(development))
-    ghc-options:        -Werror
 
 library
   import:               base, project-config

--- a/cardano-slotting/src/Cardano/Slotting/Block.hs
+++ b/cardano-slotting/src/Cardano/Slotting/Block.hs
@@ -10,6 +10,7 @@ where
 import Cardano.Binary (FromCBOR (..), ToCBOR (..))
 import Codec.Serialise (Serialise (..))
 import Control.DeepSeq (NFData)
+import Data.Aeson (ToJSON, FromJSON)
 import Data.Word (Word64)
 import GHC.Generics (Generic)
 import NoThunks.Class (NoThunks)
@@ -21,7 +22,7 @@ import Quiet (Quiet (..))
 newtype BlockNo = BlockNo {unBlockNo :: Word64}
   deriving stock (Eq, Ord, Generic)
   deriving Show via Quiet BlockNo
-  deriving newtype (Enum, Bounded, Num, Serialise, NoThunks, NFData)
+  deriving newtype (Enum, Bounded, Num, Serialise, NoThunks, NFData, ToJSON, FromJSON)
 
 instance ToCBOR BlockNo where
   toCBOR = encode

--- a/cardano-slotting/src/Cardano/Slotting/Slot.hs
+++ b/cardano-slotting/src/Cardano/Slotting/Slot.hs
@@ -3,6 +3,8 @@
 {-# LANGUAGE DeriveTraversable #-}
 {-# LANGUAGE DerivingVia #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
 
 module Cardano.Slotting.Slot
   ( SlotNo (..),
@@ -21,7 +23,7 @@ where
 import Cardano.Binary (FromCBOR (..), ToCBOR (..))
 import Codec.Serialise (Serialise (..))
 import Control.DeepSeq (NFData (rnf))
-import Data.Aeson (FromJSON, ToJSON)
+import Data.Aeson (FromJSON, ToJSON (..), Value (String))
 import Data.Typeable (Typeable)
 import Data.Word (Word64)
 import GHC.Generics (Generic)
@@ -72,6 +74,10 @@ instance NFData a => NFData (WithOrigin a) where
   rnf Origin = ()
   rnf (At t) = rnf t
 
+instance ToJSON a => ToJSON (WithOrigin a) where
+  toJSON = \case
+    Origin -> String "origin"
+    At n -> toJSON n
 
 at :: t -> WithOrigin t
 at = At

--- a/cardano-strict-containers/CHANGELOG.md
+++ b/cardano-strict-containers/CHANGELOG.md
@@ -1,3 +1,9 @@
+# Changelog for `cardano-strict-containers`
+
+# 0.1.2.1
+
+* Remove `development` flag: #372
+
 # 0.1.2.0
 
 * Added `ToCBOR` and `FromCBOR` instances for `StrictSeq`: [#361](https://github.com/input-output-hk/cardano-base/pull/361)

--- a/cardano-strict-containers/cardano-strict-containers.cabal
+++ b/cardano-strict-containers/cardano-strict-containers.cabal
@@ -1,7 +1,7 @@
 cabal-version:       >=1.10
 
 name:                cardano-strict-containers
-version:             0.1.2.0
+version:             0.1.2.1
 synopsis:            Various strict container types
 license:             Apache-2.0
 license-files:
@@ -13,11 +13,6 @@ maintainer:          operations@iohk.io
 copyright:           IOHK
 build-type:          Simple
 
-flag development
-    description: Disable `-Werror`
-    default: False
-    manual: True
-
 library
   default-language:     Haskell2010
   hs-source-dirs:       src
@@ -27,9 +22,6 @@ library
     -Wincomplete-record-updates
     -Wincomplete-uni-patterns
     -Wredundant-constraints
-  if (!flag(development))
-    ghc-options:
-      -Werror
   exposed-modules:      Data.FingerTree.Strict
                         Data.Maybe.Strict
                         Data.Sequence.Strict

--- a/heapwords/CHANGELOG.md
+++ b/heapwords/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 0.1.0.2
+
+* Remove `development` flag: #372
+
 ## 0.1.0.1
 
 * Initial release

--- a/heapwords/heapwords.cabal
+++ b/heapwords/heapwords.cabal
@@ -1,7 +1,7 @@
 cabal-version:       >=1.10
 
 name:                heapwords
-version:             0.1.0.1
+version:             0.1.0.2
 synopsis:            Heapwords
 license:             Apache-2.0
 license-files:
@@ -12,11 +12,6 @@ maintainer:          operations@iohk.io
 copyright:           IOHK
 build-type:          Simple
 extra-source-files:  CHANGELOG.md
-
-flag development
-    description: Disable `-Werror`
-    default: False
-    manual: True
 
 library
   default-language:     Haskell2010
@@ -29,9 +24,6 @@ library
     -Wredundant-constraints
     -Wunused-packages
 
-  if (!flag(development))
-    ghc-options:
-      -Werror
   exposed-modules:      Cardano.HeapWords
   build-depends:        array
                       , base

--- a/measures/CHANGELOG.md
+++ b/measures/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 0.1.0.2
+
+* Remove `development` flag: #372
+
 ## 0.1.0.1
 
 * Initial release

--- a/measures/measures.cabal
+++ b/measures/measures.cabal
@@ -1,7 +1,7 @@
 cabal-version:       >=1.10
 
 name:                measures
-version:             0.1.0.1
+version:             0.1.0.2
 synopsis:            An abstraction for (tuples of) measured quantities
 license:             Apache-2.0
 license-files:
@@ -12,11 +12,6 @@ maintainer:          operations@iohk.io
 copyright:           IOHK
 build-type:          Simple
 extra-source-files:  CHANGELOG.md
-
-flag development
-    description: Disable `-Werror`
-    default: False
-    manual: True
 
 library
   default-language:     Haskell2010
@@ -31,9 +26,6 @@ library
     -Widentities
     -Wredundant-constraints
     -Wmissing-export-lists
-  if (!flag(development))
-    ghc-options:
-      -Werror
 
   exposed-modules:
                         Data.Measure

--- a/orphans-deriving-via/CHANGELOG.md
+++ b/orphans-deriving-via/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 0.1.0.2
+
+* Remove `development` flag: #372
+
 ## 0.1.0.1
 
 * Initial release

--- a/orphans-deriving-via/orphans-deriving-via.cabal
+++ b/orphans-deriving-via/orphans-deriving-via.cabal
@@ -1,7 +1,7 @@
 cabal-version:       >=1.10
 
 name:                orphans-deriving-via
-version:             0.1.0.1
+version:             0.1.0.2
 synopsis:            Orphan instances for the base-deriving-via hooks
 license:             Apache-2.0
 license-files:
@@ -12,11 +12,6 @@ maintainer:          operations@iohk.io
 copyright:           IOHK
 build-type:          Simple
 extra-source-files:  CHANGELOG.md
-
-flag development
-    description: Disable `-Werror`
-    default: False
-    manual: True
 
 library
   default-language:     Haskell2010
@@ -31,9 +26,6 @@ library
     -Widentities
     -Wredundant-constraints
     -Wmissing-export-lists
-  if (!flag(development))
-    ghc-options:
-      -Werror
 
   exposed-modules:
                         Data.DerivingVia.DeepSeq

